### PR TITLE
[WIP] Add tutorial for Kolmogorov Equations from NeuralNetDiffEq

### PR DESCRIPTION
**Pull request by ashutosh-b-b**
1. Add a basic Fokker Planck Tutorial
2. Extend this tutorial to applications in neuronal dynamics.

### CODE REVIEW

File: tutorials/introduction/06-kolmogorov_equations.jmd [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+```julia
+xs = xspan[1]:0.01:xspan[2]
+x_val = collect(xs)
+x_val= reshape(x_val , 1 , size(x_val)[1])
+m = testmode!(m)
+y_val = m(x_val)
+y_val = reshape(y_val , length(xs) , 1)
+x_val = collect(xs)
+plot(x_val , y_val,linewidth=3,title="Solution to the linear PDE with a thick line",
```
Comment (ChrisRackauckas) [inline tutorials/introduction/06-kolmogorov_equations.jmd:52]: interesting title

File: tutorials/introduction/06-kolmogorov_equations.jmd [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            abstol=1e-10, dx = 0.001 , trajectories = 50000 ,  maxiters = 500 )
+```
+
+
+```julia
+T = tspan[2]
+mu = 1
+sig = 0.5
+sig2 = (sig^2 /(2*mu))exp(-2*mu*T)*(exp(2*mu*T) - 1)
+analytical(xi) = pdf.(Normal(0.00 , sqrt(sig2)) , xi)
```
Comment (ChrisRackauckas) [inline tutorials/introduction/06-kolmogorov_equations.jmd:39]: are you sure about the scaling on the input here? IIRC it was supposed to be the variance.

Comment (ashutosh-b-b) [reply tutorials/introduction/06-kolmogorov_equations.jmd:39]: Where is the input scaled? Its sigma^2 and then I have taken its sqrt in Normal Pdf.

Comment (ashutosh-b-b) [reply tutorials/introduction/06-kolmogorov_equations.jmd:39]: Screen Shot 2020-06-10 at 12 19 35 PM
This is supposed to be the pdf right?

Comment (ChrisRackauckas) [reply tutorials/introduction/06-kolmogorov_equations.jmd:39]: With Distributions.jl you're supposed to do Normal(mean,variance) IIRC, and you're passing Normal(mean,std)?

File: tutorials/introduction/06-kolmogorov_equations.jmd [hunk 3]
```diff
@@ -0,0 +1,75 @@
+---
+title: Kolmogorov Equations
```
Comment (ChrisRackauckas) [inline tutorials/introduction/06-kolmogorov_equations.jmd:2]:
```suggestion
title: Kolmogorov Backward Equations
```

File: tutorials/introduction/06-kolmogorov_equations.jmd [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+author: Ashutosh Bharambe
+---
+
+```julia
+using Flux, StochasticDiffEq
+using  NeuralNetDiffEq
+using Plots
+using Distributions
+```
+## Introduction on Kolmogorov Equations
```
Comment (ChrisRackauckas) [inline tutorials/introduction/06-kolmogorov_equations.jmd:12]: backward. I'd always mention it's the backward equation for clarity.

File: tutorials/introduction/06-kolmogorov_equations.jmd [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+The one dimensional backward kolmogorov equation that we are going to deal with is of the form :
+$$
+\begin{equation}
+  \frac{\partial p}{\partial t} = \mu(x)\frac{\partial p}{\partial x} + \frac{1}{2}{\sigma^2}(x)\frac{\partial^2 p}{\partial x^2}
+\end{equation}
+$$
+Consider a Black Scholes Partial Differential Equation of the form :
+$$
+\begin{equation}
+  \frac{\partial V}{\partial t} + rS\frac{\partial V}{\partial S} + \frac{1}{2}{\sigma^2}{S^2}\frac{\partial^2 V}{\partial S^2} -rV = 0
```
Comment (ChrisRackauckas) [inline tutorials/introduction/06-kolmogorov_equations.jmd:23]: This is missing some important details. It's a terminal PDE, so it's defined with valuations on the end. How does this relate to the model? Describe the options market and where this idea comes from.

File: tutorials/introduction/06-kolmogorov_equations.jmd [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+\end{equation}
+$$
+
+In order to make the above equation in the form of the Backward - Kolmogorov PDE we should substitute
+
+$$
+\begin{equation}
+  V(S , t) = e^{rt}p(S , t)
```
Comment (ChrisRackauckas) [inline tutorials/introduction/06-kolmogorov_equations.jmd:31]: demonstrate that this substitution works by showing what the PDE transforms into

**Comment by ChrisRackauckas** [comment]
Overall very good start, but it's missing some of those pieces in the introduction to explain and motivate the model.

File: tutorials/introduction/06-kolmogorov_equations.jmd [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+\end{equation}
+$$
+Consider a Black Scholes Partial Differential Equation of the form :
+The Black-Scholes Model governs the price evolution of the European put or call option. In the below equation V is the price of some derivative , S is the Stock Price , r is the risk free interest
+rate and σ the volatility of the stock returns. The payoff at a time T is known to us. And this makes it a terminal PDE.
+$$
+\begin{equation}
+  \frac{\partial V}{\partial t} + rS\frac{\partial V}{\partial S} + \frac{1}{2}{\sigma^2}{S^2}\frac{\partial^2 V}{\partial S^2} -rV = 0
+\end{equation}
+$$
```
Comment (ChrisRackauckas) [inline tutorials/introduction/06-kolmogorov_equations.jmd:28]: where... you're not giving the full definition of the PDE. You need to define that it has a valuation at the end.

Comment (ashutosh-b-b) [reply tutorials/introduction/06-kolmogorov_equations.jmd:28]: Do I need to explain what is a PDE? or the Black Scholes PDE in detail?

Comment (ChrisRackauckas) [reply tutorials/introduction/06-kolmogorov_equations.jmd:28]: No, not "a PDE", "the PDE". You haven't defined the well-defined PDE, in the mathematical sense of "well-defined". There's no domain. If you don't define the terminal condition, the PDE isn't solvable.

**Comment by ChrisRackauckas** [comment]
Merging assuming it builds properly.

**Pull request merged**